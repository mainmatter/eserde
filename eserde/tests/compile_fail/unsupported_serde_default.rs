#[derive(eserde::Deserialize)]
struct Foo {
    #[serde(rename = "route", default = "default_route")]
    route_0: String,
    #[serde(default = "default_route")]
    route_1: String,
}

fn default_route() -> String {
    "/".to_string()
}

fn main() {}
