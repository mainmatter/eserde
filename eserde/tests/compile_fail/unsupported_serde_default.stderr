error: `eserde::Deserialize` doesn't yet support the `#[serde(default = "..")]` attribute. It only supports `#[serde(default)]`, which defers to the `Default` trait to generate the default value.
 --> tests/compile_fail/unsupported_serde_default.rs:3:31
  |
3 |     #[serde(rename = "route", default = "default_route")]
  |                               ^^^^^^^^^^^^^^^^^^^^^^^^^

error: `eserde::Deserialize` doesn't yet support the `#[serde(default = "..")]` attribute. It only supports `#[serde(default)]`, which defers to the `Default` trait to generate the default value.
 --> tests/compile_fail/unsupported_serde_default.rs:5:13
  |
5 |     #[serde(default = "default_route")]
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^
